wordpress-mobile/WordPress-Android
# Update visibility of scan menu item

Parent issue #13326 

This PR uses "JetpackCapabilityUseCase" instead of "ScanStatusService" to find out if the user has permissions to use Scan feature. It implements the logic for both legacy and improved MySiteFragment.

The result of /capabilities endpoint is stored into shared preferences and is kept as valid for 15 minutes.


Merge instructions:
1. Merge [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1831)
2. Replace fluxc hash with tag
3. Remove "Not ready for merge" label
4. Merge this PR


To test:
1. Disable "Scan" feature flag
2. Open My Site (make sure a jetpack site with scan feature is selected)
3. Verify "Scan" menu item is hidden
-------
1. Enable "Scan" feature flag
2. Open My site (make sure a jetpack site with scan feature is selected)
3. Verify "Scan" menu item is visible
4. Switch to a site which doesn't have Scan capability
5. Make sure the Scan menu item is hidden
------------
1. Enable "Scan" feature flag
2. Enable "ImproveMySite" feature flag
3. Open My Site (make sure a jetpack site with scan feature is selected)
4. Verify "Scan" menu item is visible
5. Switch to a site which doesn't have Scan capability
6. Make sure Scan menu item is hidden
7. Switch to a site with Backup feature
8. Open Activity Log
9. Make sure the filters are shown

PR submission checklist:
